build-logic/src/main/kotlin/at/asitplus/gradle/SignumConventionsPlugin.kt

```diff
@@ -0,0 +1,332 @@
+package at.asitplus.gradle
+
+import com.android.build.api.dsl.androidLibrary
+import com.android.build.api.variant.KotlinMultiplatformAndroidComponentsExtension
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.tasks.StopExecutionException
+import org.gradle.api.tasks.testing.Test
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.kotlin.dsl.*
+import org.gradle.plugins.signing.SigningExtension
+import org.jetbrains.kotlin.gradle.ExperimentalWasmDsl
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
+import org.jetbrains.kotlin.konan.target.Family
+import java.io.ByteArrayOutputStream
+
+/**
+ * Gradle convention plugin for Signum
+ */
+class SignumConventionsPlugin : Plugin<Project> {
+    override fun apply(target: Project) = with(target) {
+        logger.info("SignumC onventions Plugin applied to project: ${'$'}{target.path}")
+        pluginManager.apply("org.jetbrains.kotlin.multiplatform")
+        pluginManager.apply("org.jetbrains.kotlin.plugin.serialization")
+        pluginManager.apply("com.android.kotlin.multiplatform.library")
+        pluginManager.apply("signing")
+        pluginManager.apply("at.asitplus.gradle.conventions")
+        pluginManager.apply("de.infix.testBalloon")
+    }
+}
+
+class SignumConventionsExtension(private val project: Project) {
+    init {
+        val indispensableVersion: String by project.extra
+        project.version = indispensableVersion
+        //if we do this properly, cinterop (swift-klib) blows up, so we hack!
+        project.afterEvaluate {
+            tasks.withType<Test>().configureEach {
+                maxHeapSize = "4G"
+            }
+        }
+        project.silence()
+        project.fermentRottenApples()
+
+        project.extensions.getByType<KotlinMultiplatformExtension>().apply {
+            compilerOptions.freeCompilerArgs.add("-Xexpect-actual-classes")
```

Comment:
add the opt in flag for mandatory return type checking maybe? might be good as a separate PR/issue